os/Makefile.unix

```diff
@@ -705,3 +705,18 @@ tools/compression/mkcompresstool.sh: include/tinyara/config.h
 ifeq ($(CONFIG_COMPRESSED_BINARY),y)
 	$(Q) tools/compression/mkcompresstool.sh
 endif
+
+# ../tools/fota_delta/mkfotadeltatool.sh
+#
+# This target will make a compression tool by calling mkcompresstool.sh.
+# This compression tool (mkcompressimg) will be used to compress binaries.
+# Hence, this tool needs to be created before we build those binaries.
+
+../tools/fota_delta/mkfotadeltatool.sh: include/tinyara/config.h
+ifeq ($(CONFIG_COMPRESSED_BINARY),y)
+	$(Q) ../tools/fota_delta/mkfotadeltatool.sh
+else
+ifeq ($(CONFIG_FOTA_DELTA_ENGINE),y)
+	$(Q) ../tools/fota_delta/mkfotadeltatool.sh
```

Comment:
Why do you build it through shell script even clean command is executed through makefile?